Fachportal Padagogik.js

```diff
@@ -102,31 +104,15 @@ function scrape(doc, url) {
 		});
 	} else {
 		var pdfUrl = ZU.xpathText(doc, '//meta[@name="citation_pdf_url"]/@content');
-		Z.debug(pdfUrl);
-
-		/*
-		//fails on http://www.pedocs.de/frontdoor.php?source_opus=9491&la=de
-		Z.debug(ZU.xpathText(doc, '//span[@class="Z3988"]/@title'));
-		var translator = Zotero.loadTranslator("web");
-		translator.setTranslator("05d07af9-105a-4572-99f6-a8e231c0daef");//COinS translator
-		translator.setDocument(doc);//TODO encoding is normally Western and not utf8
-		translator.setHandler("itemDone", function (obj, item) {
-			if (pdfUrl) {
-				item.attachments.push({
-					url: pdfUrl,
-					title: "Full Text PDF",
-				});
-			}
-			
-			finalize(doc, item);
-		});
-		translator.translate();
-		*/
+		//Z.debug(pdfUrl);
 		
 		var coins = ZU.xpathText(doc, '//span[@class="Z3988"]/@title');
 		if (coins) {
+			coins = coins.replace(/%26%238208%3B/g, '-')
+				.replace(/%C2%82/g, '‚').replace(/%C2%91/g, '‘');
```

Comment:
I'd prefer to do this in a more generic fashion.
>&#8208; instead of - (derived from %26%238208%3B)

is just a matter of applying `ZU.unescapeHTML` on all fields in the resulting item after COinS is parsed.

>� instead of ‚ (derived from %C2%82)

Is a bit tricky. Since this was Windows-1252 encoding that was interpreted as UTF-8 and then URI-encoded. But here's something I cooked up that should work in all cases.
```
/**
 * Transcodes non-UTF-8-encoded text that was passed through encodeURIComponent
 * (which assumed that it was UTF-8) into the UTF-8 equivalent. This makes decodeURIComponent
 * correctly decode the encoded text into UTF-8.
 *
 * E.g. 0x82 (SINGLE LOW-9 QUOTATION MARK) in Windows-1252 is treated as UTF-8 and encoded
 * as %C2%82. This would transcode it to %E2%80%9A and decodeURIComponent would then
 * decode it to "\u201A" (the same symbol)
 *
 * @param {String} s String to transcode
 * @param {String} fromEncoding String specifying source encoding. Only "Windows-1252" is
 *     currently supported.
 * @return {String} Transcoded string. Unaltered string is returned if the encoding is not recognized.
 * 
 * Examples:
 * transcodeURIEncoding("%C2%82", "Windows-1252") => "%E2%80%9A"
 * transcodeURIEncoding("%26%238208%3B", "Windows-1252") => "%26%238208%3B"
 */
function transcodeURIEncoding(s, fromEncoding) {
	// Only differing code point need to be specified
	var map = {
		// http://unicode.org/Public/MAPPINGS/VENDORS/MICSFT/WINDOWS/CP1252.TXT
		'Windows-1252': {
			0x80: 0x20AC,
			0x81: null,
			0x82: 0x201A,
			0x83: 0x0192,
			0x84: 0x201E,
			0x85: 0x2026,
			0x86: 0x2020,
			0x87: 0x2021,
			0x88: 0x02C6,
			0x89: 0x2030,
			0x8A: 0x0160,
			0x8B: 0x2039,
			0x8C: 0x0152,
			0x8D: null,
			0x8E: 0x017D,
			0x8F: null,
			0x90: null,
			0x91: 0x2018,
			0x92: 0x2019,
			0x93: 0x201C,
			0x94: 0x201D,
			0x95: 0x2022,
			0x96: 0x2013,
			0x97: 0x2014,
			0x98: 0x02DC,
			0x99: 0x2122,
			0x9A: 0x0161,
			0x9B: 0x203A,
			0x9C: 0x0153,
			0x9D: null,
			0x9E: 0x017E,
			0x9F: 0x0178
		}
	}[fromEncoding];

	if (!map) return s;

	return s.replace(/%([CD][0-9A-F])%([0-9A-F]{2})/g, function(m, c1, c2) {
		var codePoint = ((parseInt(c1, 16) % 0b11000000) << 6)
			+ parseInt(c2, 16) % 0b10000000;

		if (!(codePoint in map)) return m;

		codePoint = map[codePoint];
		if (codePoint === null) {
			// Code point was not defined. Shouldn't really happen, unless
			// an incorrect fromEncoding was specified.
			return '';
		}

		return encodeURIComponent(String.fromCodePoint(codePoint));
	});
}
```

At some point, it may be worth porting this to ZU, or some other shared component.